### PR TITLE
chore: fix library duplication warnings for macos builds

### PR DESCRIPTION
**Pull request by antonbaliasnikov**
# What ❔

Disable duplicate library warnings on XCode >= 15.

## Why ❔

To have clean output logs.

## Checklist

- [x] PR title corresponds to the body of PR.
- [x] Documentation comments have been added / updated.
- [x] Code has been formatted via `cargo fmt` and checked with `cargo clippy`.

### CODE REVIEW

File: src/platforms/shared.rs [hunk 1]
```diff
@@ -82,3 +81,21 @@ pub fn shared_build_opts_ccache(use_ccache: bool) -> Vec<String> {
         vec![]
     }
 }
+
+/// Ignore duplicate libraries warnings for MacOS with XCode>=15.
+pub fn macos_build_opts_ignore_dupicate_libs_warnings() -> Vec<String> {
+    let xcode_version =
+        crate::utils::get_xcode_version().unwrap_or(crate::utils::XCODE_MIN_VERSION);
+    if xcode_version >= 15 {
+        println!(
```
Comment (hedgar2017) [inline src/platforms/shared.rs:90]: `eprintln!` please!

File: src/platforms/shared.rs [hunk 2]
```diff
@@ -82,3 +81,21 @@ pub fn shared_build_opts_ccache(use_ccache: bool) -> Vec<String> {
         vec![]
     }
 }
+
+/// Ignore duplicate libraries warnings for MacOS with XCode>=15.
```
Comment (hedgar2017) [inline src/platforms/shared.rs:85]: Let's use surrounding `///` lines for functions.

File: Cargo.toml [hunk 3]
```diff
@@ -24,6 +24,7 @@ toml = "0.7"
 num_cpus = "1.15"
 fs_extra = "1.2"
 path-slash = "0.2"
+regex = "1.10.4"
```
Comment (hedgar2017) [inline Cargo.toml:27]: Please keep minor selector like above.

File: src/platforms/shared.rs [hunk 4]
```diff
@@ -82,3 +81,21 @@ pub fn shared_build_opts_ccache(use_ccache: bool) -> Vec<String> {
...
+pub fn macos_build_opts_ignore_dupicate_libs_warnings() -> Vec<String> {
+    let xcode_version =
+        crate::utils::get_xcode_version().unwrap_or(crate::utils::XCODE_MIN_VERSION);
+    if xcode_version >= 15 {
```
Comment (hedgar2017) [inline src/platforms/shared.rs:89]: Let's move this magic number to constant as well.

**Review by hedgar2017 — CHANGES_REQUESTED** [review]

File: src/utils.rs [hunk 5]
```diff
@@ -1,14 +1,18 @@
 //!
-//! The zkEVM LLVM builder utilities.
+//! The LLVM builder utilities.
```
Comment (FlashSheridan) [inline src/utils.rs:2]: I don’t think we want to remove all Matter Labs branding and look like a generic LLVM product, and `zkevm-llvm` is still the binary name, though the repo name is now era-compiler-llvm-builder.

Comment (hedgar2017) [reply src/utils.rs:2]: Yeah, renaming the binary is tricky, but we should gradually make it target-independent.

**Review by hedgar2017 — CHANGES_REQUESTED** [review]
Please turn off failfast everywhere!

File: src/platforms/shared.rs (hunk 1 above)
Comment (antonbaliasnikov) [reply src/platforms/shared.rs]: This is not an error, this is an info message. But I believe we don't print anything like that in our tool, so, I just removed this.

File: src/platforms/shared.rs (hunk 2 above)
Comment (antonbaliasnikov) [reply src/platforms/shared.rs:94]: Done.

File: Cargo.toml (hunk 3 above)
Comment (antonbaliasnikov) [reply Cargo.toml]: Done.

File: src/platforms/shared.rs (hunk 4 above)
Comment (antonbaliasnikov) [reply src/platforms/shared.rs]: Done.

File: src/utils.rs (hunk 5 above)
Comment (antonbaliasnikov) [reply src/utils.rs:2]: > but we should gradually make it target-independent.

That's interesting! @hedgar2017 how do you see this in general?

File: tests/common/mod.rs [hunk 6]
```diff
@@ -4,7 +4,7 @@ pub const ZKEVM_LLVM: &str = "zkevm-llvm";
 pub const PACKAGE_VERSION: &str = env!("CARGO_PKG_VERSION");
 pub const ERA_LLVM_REPO_URL: &str = "https://github.com/matter-labs/era-compiler-llvm";
 pub const ERA_LLVM_REPO_TEST_BRANCH: &str = "v1.4.2";
-pub const ERA_LLVM_REPO_TEST_REF: &str = "714cc6aa05ea03dab3e4e504cb8c25973a33ea94";
+pub const ERA_LLVM_REPO_TEST_REF: &str = "b5ccf6d5774e9bc3cee47ab4a334404718d3adfd";
```
Comment (antonbaliasnikov) [inline tests/common/mod.rs:7]: Annoying issue for now... but I would fix it separately with a CI update to unify matrix jobs as part of [CPR-1674](https://linear.app/matterlabs/issue/CPR-1674/make-compiler-builder-unit-tests-independent-from-the-commit)

File: src/utils.rs (hunk 5 above)
Comment (hedgar2017) [reply src/utils.rs:2]: @antonbaliasnikov I believe it's just about naming, so we should get rid of targets (especially their outdated names like `zkevm`) and only brand this tool as Matter Labs LLVM builder :)

**Review by hedgar2017 — APPROVED** [review]

File: src/platforms/shared.rs (hunk 1 above)
Comment (hedgar2017) [reply src/platforms/shared.rs]: @antonbaliasnikov yeah, that may sound counter-intuitive.
In our binaries we usually print all diagnostic messages and logs to stderr, and reserve stdout for output with a potentially pre-defined format, with the most common case as solc/zksolc standard JSON I/O.
So the parent process can always parse stdout, and simply bubble-up stderr if exit code is non-zero.

File: src/platforms/shared.rs (hunk 2 above)
Comment (FlashSheridan) [reply src/platforms/shared.rs:94]: Please don’t merge all commits into one while the review is still in process.  Clicking on the “[View it on GitHub](https://github.com/matter-labs/era-compiler-llvm-builder/pull/24/files/d1c2a61ffad69c2852af119fcc7c3daa61ec8257..afc751850972f2796a5281455fd9ecbd4df6e195)” link in the notification email now gives:

> We went looking everywhere, but couldn’t find those commits.
> Sometimes commits can disappear after a force-push.

It’s a lot easier to review changes purely in comments than the whole thing over again.  :-) 

(Eventually I found them via the [hash link](https://github.com/matter-labs/era-compiler-llvm-builder/commit/afc751850972f2796a5281455fd9ecbd4df6e195) in the email:  “This commit does not belong to any branch on this repository, and may belong to a fork outside of the repository.”)

**Pull request merged**